### DISCUSSION TITLE
flashing ESP32

### DISCUSSION THREAD

**Discussion by misu1lica** [emsesp/EMS-ESP32#1045] [General]
Hi
I have ESP32 WROOM-32UE and I flash with "flash_download_tool_v3.8.5" version EMS-ESP-2_1_0-esp32.bin.
Programming is OK, but when I restart ESP32, the sistem don't have a Acces point esp-ems.
On the RS232 I see the butloader information.
What can I do?

**Comment by proddy** [comment]
So no AP is seen in your wifi router? Also do you see any LEDs flashing?

I'm not familiar with the download tool. Do you have a link to it?

**Comment by misu1lica** [comment]
The link: 

https://www.espressif.com/sites/default/files/tools/flash_download_tool_v3.8.5.zip
Yes, I don't seen AP.
This module don't have LED on board, I try to put one.

**Comment by misu1lica** [comment]
Hi,
 the LED does not flashing. 
The program can't program ESP32 WROOM-32UE?
I try to program with uPyCraft_V1.1.exe  and the same, the ESP did not have AP

On serial I read this:

rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
flash read err, 1000
ets_main.c 371 
ets Jun  8 2016 00:22:57

**Comment by proddy** [comment]
If you build the standard 'blink' app does it work?

If you're new to building and uploading, start with PlatformIO and pick the right flash mode for your board (dio/dout).

**Comment by misu1lica** [comment]
I programm esp8266 and all is OK.
But with esp32 don't work.

**Comment by proddy** [comment]
I can't help you without more information, sorry.

**Comment by misu1lica** [comment]
Hi
use ESPHome-Flasher-1.2.0-windows-x64 , https://github.com/esphome/esphome-flasher/releases work for ESP32!
But i must put in flash mode my ESP32 but press RST+BOOT, then release RST and after 1 secund relase BOOT.
And another schematic
![ems-esp2](https://user-images.githubusercontent.com/22762676/99358524-5202b480-28b6-11eb-82b3-2fa9d4305e4a.jpg)
misu

**Comment by proddy** [comment]
@misu1lica impressive design. Can you share some photos and also how you intend to use it? specially those photocouplers

**Comment by misu1lica** [comment]
I started with the motherboard and now I'm starting to program ESP.
For OK4 I want to monitor the boiler supply voltage (230Vac). We put power plants on the UPS - we still have voltage interruptions.
For OK3 I want to start a boiler.
For OK5 voltage drop alarm ...
